### PR TITLE
implementation method moveRobot() in class FieldXY

### PR DESCRIPTION
**Pull request by KrohIgor**

### CODE REVIEW

**Comment by KrohIgor** [comment]
Please tell me which of the commits is better?

File: src/main/java/core/basesyntax/FieldXY.java [hunk 1]
```diff
@@ -13,7 +13,57 @@
...
+    private void moveRobotForX(Robot robot, int toX) {
+        if (robot.getX() < toX && !robot.getDirection().equals(Direction.RIGHT)) {
+            turnRobot(robot, Direction.RIGHT);
+        } else if (!robot.getDirection().equals(Direction.LEFT)) {
+            turnRobot(robot, Direction.LEFT);
+        }
+        while (robot.getX() != toX) {
+            robot.stepForward();
+        }
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FieldXY.java:44]: What the difference in logic between these methods?

File: src/main/java/core/basesyntax/FieldXY.java [hunk 2]
```diff
@@ -13,7 +13,57 @@
...
+                && robot.getDirection().equals(Direction.DOWN))
+                || (direction.equals(Direction.UP)
+                && robot.getDirection().equals(Direction.RIGHT))
+                || (direction.equals(Direction.DOWN)
+                && robot.getDirection().equals(Direction.LEFT))) {
+            robot.turnLeft();
+        } else {
+            robot.turnRight();
+        }
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FieldXY.java:68]: Try to optimize it.

Comment (KrohIgor) [reply src/main/java/core/basesyntax/FieldXY.java]: I did not understand the choice of turning right or left. I wanted to make a smart robot, and not just make it spin in place.

**Pull request closed without merging**